src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs

```diff
@@ -590,6 +581,8 @@ public Task(Action<object> action, object state, CancellationToken cancellationT
 
                 AssignCancellationToken(cancellationToken, null, null);
             }
+
+            PossiblyCaptureContext();
```

Comment:
I would inline it here, so that it looks just like CoreCLR version. It is not really worth it to have it in a separate function when it is called just from a single place.

Reply:
PossiblyCaptureContext seems to be called from `Future.cs` and `TaskContinuation.cs`, which is why I kept it as-is to avoid larger refactoring.